src/SAML2/XML/samlp/StatusDetail.php

```diff
@@ -64,7 +98,11 @@ public function setDetail(?\DOMElement $detail): void
      */
     public static function fromXML(DOMElement $xml): object
     {
-        return new self($xml);
+        Assert::same($xml->localName, 'StatusDetail');
+        Assert::same($xml->namespaceURI, Constants::NS_SAMLP);
+
+        /** @psalm-var \DOMElement $xml->childNodes[1] */
+        return new self([new Chunk($xml->childNodes[1])]);
```

Comment:
This is a bug. We should iterate over the `DOMNodeList` in `$xml->childNodes` and create a new `SAML2\XML\Chunk` for each `DOMElement`, then pass an array of all those chunks to the constructor.

Reply:
Has been resolved